src/pybind/ceph_argparse.py

```diff
@@ -341,9 +341,8 @@ def __init__(self, goodchars=''):
 
     def valid(self, s: str, partial: bool = False) -> None:
         sset = set(s)
-        if self.goodset and not sset <= self.goodset:
-            raise ArgumentFormat("invalid chars {0} in {1}".
-                                 format(''.join(sset - self.goodset), s))
+        if self.goodset and not re.match(self.goodchars, s):
```

Comment:
It's unclear if this is just an improvement or a bug fix that you identified. Could you explain a bit more?

Reply:
dropped this change in latest PR update